server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -322,32 +321,42 @@ void executeRequest(
             relativeStartNanos,
             System::nanoTime
         );
+
+        final ClusterState clusterState = clusterService.state();
+        clusterState.blocks().globalBlockedRaiseException(ClusterBlockLevel.READ);
```

Comment:
@Mikep86 Can you remind me why this check for clock was added as part of this PR?

Reply:
I see you did not add it, it was there before, you just moved it :)

Reply:
If I remember correctly, this needed to be moved here because we need to check it before resolving the indices

Reply:
sounds good, thanks